src/main/java/scripts/versionscripts/ReadFromTerraHelmfileRepo.java

```diff
@@ -29,75 +28,49 @@ public ReadFromTerraHelmfileRepo() {}
    * @param parameters list of string parameters supplied by the version list
    */
   public void setParameters(List<String> parameters) throws Exception {
-    if (parameters == null || parameters.size() < 2) {
+    if (parameters == null || parameters.size() < 3) {
       throw new IllegalArgumentException(
           "Must provide terra helmfile file paths in the parameters list");
     }
-    baseFilePath = parameters.get(0);
-    overrideFilePath = parameters.get(1);
+    appName = parameters.get(0);
+    baseFilePath = parameters.get(1);
+    overrideFilePath = parameters.get(2);
   }
 
   /**
    * This method determines the version by reading the helm app/chart versions in the terra-helmfile
    * GitHub repository.
    */
   public VersionScriptResult determineVersion(ServerSpecification server) throws Exception {
-    // TODO QA-1643: Re-enable importComponentVersions API route pending DevOps readiness
-    // Map<String, Map<String, String>> kubernetesComponentVersions =
-    //    !server.skipKubernetes ? KubernetesClientUtils.importComponentVersions() : null;
+    return new VersionScriptResult.Builder()
+        .helmVersions(buildHelmVersion(server, appName, baseFilePath, overrideFilePath))
+        .build();
+  }
 
+  public static List<HelmVersion> buildHelmVersion(
+      ServerSpecification server, String appName, String baseFilePath, String overrideFilePath)
+      throws Exception {
     // Pull versions from terra-helmfile
-    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
-    mapper.findAndRegisterModules();
-    HelmRelease helmRelease = mapper.readValue(new File(baseFilePath), HelmRelease.class);
-    HelmRelease helmOverride = mapper.readValue(new File(overrideFilePath), HelmRelease.class);
+    HelmRelease helmRelease = HelmRelease.fromFile(baseFilePath);
+    HelmRelease helmOverride = HelmRelease.fromFile(overrideFilePath);
 
     // Loop through the override helm values and replacing any base helm values, then returning the
     // now-modified base helm values map.
-    merge(helmOverride.getReleases(), helmRelease.getReleases());
+    HelmRelease.merge(helmOverride, helmRelease);
     logger.info(
         "Done merging Helm override values from {} into base values obtained from {}",
         overrideFilePath,
         baseFilePath);
 
-    // Pull 'workspacemanager' versions from terra-helmfile, add more service versions if needed.
-    String wsmHelmAppVersion =
-        helmRelease.getReleases().get("workspacemanager").getAppVersion().orElse("");
-    String wsmHelmChartVersion =
-        helmRelease.getReleases().get("workspacemanager").getChartVersion().orElse("");
+    // Pull appName versions from terra-helmfile, add more service versions if needed.
+    String helmAppVersion = helmRelease.getReleases().get(appName).getAppVersion().orElse("");
```

Comment:
nit: perhaps log if the appName/appVersion is not found?

Reply:
Added log statements for empty app or chart version.